testdata/parenthesis/checker_tests.go

```diff
@@ -7,3 +7,15 @@ func badReturn() [](func()) {
 func veryBadReturn() [](func([](func()))) {
 	return nil
 }
+
+var _ [](func())
+var _ [5](*int)
+var _ [](func())
+
+var (
+	_ int
+	_ [5](*int)
+	_ [](func())
+)
+
+const _ (int) = 5
```

Comment:
Add 1 type decl here and make sure that checker does not panic.
We will merge after that and you can continue to work on `ast.TypeSpec` in a separate PR.